.circleci/config.yml

```diff
@@ -0,0 +1,145 @@
+# FUN packaging CI
+version: 2
+
+# List jobs that will be integrated and executed in a workflow
+jobs:
+  # Build application package
+  build:
+    docker:
+      - image: circleci/python:2.7-stretch
+
+    steps:
+      - checkout
+
+      # Restore dependencies cache only if nothing has changed in the
+      # `setup.cfg` file. There is no fall-back to older virtual environments to
+      # ensure a clean environment if some dependencies may have been upgraded
+      # or removed.
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "setup.cfg" }}
```

Comment:
with this strategy I'm wondering if we can miss a dependency of our dependencies being upgraded and breaking our code?

Reply:
If we pin our primary dependencies, secondary dependencies won't be upgraded. If we upgrade primary dependencies and the CI fails (because #4 has been implemented), we've earned a :medal_sports:.

Are you saying that you would prefer not caching dependencies at all?